### DISCUSSION TITLE
How to get a Collection to be Global

### DISCUSSION THREAD

**Discussion by DSchaff** [tinacms/tinacms#4556] [Help] [answered]
I can't figure out if I'm doing something wrong or it's broken, but I can't get a Collection to be global and show in the Site list.  I set ui.global = true in the collection definition and it's still treating it as a normal Collection, showing in the Collections list and not the Site list in the sidebar.

What I'm doing:  
1) run npx create-tina-app@latest and choose the barebones starter
2) find the Blog Post collection (tina/collections/post.js) and edit the "ui" property to remove the "router" property and instead add a "global" property set to true.
3) check the admin sidebar and the Blog Posts are still in Collections list and not in Site list.

**Comment by guillermodlpa** [comment]
Came across this today. I'm wondering a similar thing. For me, it always shows in Collections, and sometimes in Site also

**Reply by DSchaff** [reply]
See my other comment for the answer, but in your case I'd imagine it does show in the Site section when you are on a page that called the useTina hook on that collection, and it doesn't show for pages where you didnt.

If you want to filter them out of the Collections list, you might be able to take advantage of the cmsCallback in your defineConfig options and mess with cms.api.admin.fetchCollections, which is the function that returns the list of collections.  I just set it equal to () => [] because I don't want anything in the Collection list, but for you maybe you can preserve the original fetchCollections in a variable, and then overwrite cms.api.admin.fetchCollections with a function that calls the original function and then filters the results based on the collection name.

**Reply by guillermodlpa** [reply]
Thanks @DSchaff ! I'll investigate. So far, no luck. I think `fetchCollections` seems to be called before the `cmsCallback` in some instances, so any overrides don't apply. I'll keep digging.


I've seen though that this issue I describe happens as well in the [Tina cloud starter repo.](https://github.com/tinacms/tina-cloud-starter), so maybe one day me or someone else can figure it out and fix it for good.

**Comment by DSchaff** [comment, marked as answer]
I figured it out, it's because I wasn't calling the useTina hook on the global collection in each page component.  

I don't see that listed as a requirement anywhere that "global" is mentioned in the Collection documentation, and it doesn't make sense that would be required because a major use-case of a global collection is for it to always be accessible and editable from the sidebar without having to be linked to each/any of your visually-editable pages.
